### PR TITLE
Cart management (ALL P0 Features Done)

### PR DESCRIPTION
**Pull request by ChhikaraBRUH**
Preview Link: https://deploy-preview-5--snkrs-store.netlify.app/

/ All P0 Features of E Comm Completed /

- Added front page categories to filters on product listing page
- Added Cart Management
- Linked Cart and Wishlist Pages
- Linked Product, Wishlist, Cart Cards everywhere
- Added Filters responsiveness

### CODE REVIEW

**Comment by Jay-Gadhiya** [comment]
- ui working fine and looks good

File: src/context/cart/cartContext.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import axios from "axios";
+import React, { createContext, useContext, useReducer } from "react";
+import { cartInitialState, cartReducer } from "../../reducer/cartReducer";
+import { useAuth } from "../auth/authContext";
+
+const CartContext = createContext(cartInitialState);
+
+const getCartContextValue = () => {
+	const [cartState, cartDispatch] = useReducer(cartReducer, cartInitialState);
```
Comment (Jay-Gadhiya) [inline src/context/cart/cartContext.js:9]: I like your approach to this function, you can extract this into a new file also.

File: src/pages/Home/components/Category/Category.jsx [hunk 2]
```diff
@@ -57,7 +67,14 @@ const Category = () => {
 
 							<div className='card-footer'>
 								<Link to={"/products"}>
-									<button className='card-button'>Check It Out</button>
+									<button
+										className='card-button'
+										onClick={async () => {
+											await filterDispatch({ type: "CLEAR_ALL" });
+											filterDispatch({ type: "RUNNING" });
+										}}>
```
Comment (Jay-Gadhiya) [inline src/pages/Home/components/Category/Category.jsx:75]: You can write this function on top of the return, for batter code readability.

File: src/pages/Home/components/Category/Category.jsx [hunk 3]
```diff
@@ -83,7 +100,14 @@ const Category = () => {
 
 							<div className='card-footer'>
 								<Link to={"/products"}>
-									<button className='card-button'>Check It Out</button>
+									<button
+										className='card-button'
+										onClick={async () => {
+											await filterDispatch({ type: "CLEAR_ALL" });
+											filterDispatch({ type: "SB" });
+										}}>
```
Comment (Jay-Gadhiya) [inline src/pages/Home/components/Category/Category.jsx:108]: here also

**Review by Jay-Gadhiya — APPROVED** [review]
code indentations are good.
good job @ChhikaraBRUH

**Review by RahullRawat — APPROVED** [review]

**Pull request merged**